cli/src/commands/operation/undo.rs

```diff
@@ -75,6 +76,16 @@ pub fn cmd_op_undo(
         let template = tx.base_workspace_helper().operation_summary_template();
         template.format(&bad_op, formatter.as_mut())?;
         writeln!(formatter)?;
+
+        let is_bare_undo =
+            command.matches().subcommand_matches("operation").is_none() && args.operation == "@";
+        let was_undo_operation = bad_op.metadata().description.starts_with("undo operation");
```

Comment:
I'm not entirely happy with this check here, but it does work (as long as the operation description remains consistent).

A better alternative might be to attach metadata to each operation that indicates whether that operation is an undo.  I see there is the space for arbitrary key-value pairs in the metadata, presumably I could use that?  For now I've chosen to take the easier option, though, to keep the scope of this PR small.

Reply:
Yes, we've talked about adding metadata. For now, I think we can instead compare the view objects:

```rust
let was_undo_operation = tx.repo().view() == parent_op.view();
```